java/parser/Parser.java

```diff
@@ -897,25 +899,42 @@ public Object visitValue(GraqlParser.ValueContext ctx) {
             return getDateTime(ctx.DATETIME_());
 
         } else {
-            throw new IllegalArgumentException("Unrecognised Literal token: " + ctx.getText());
+            throw GraqlException.of(PARSING_ERROR_LITERAL_TOKEN.message(ctx.getText()));
         }
     }
 
     private String getString(TerminalNode string) {
-        // Remove surrounding quotes
-        return unquoteString(string);
+        if (string.getText().length() >= 2) {
+            final GraqlToken.Char literalFirstSymbol = GraqlToken.Char.of(string.getText().substring(0,1));
+            final GraqlToken.Char literalLastSymbol = GraqlToken.Char.of(string.getText().substring(string.getText().length() - 1));
+            if (literalFirstSymbol != null && literalFirstSymbol.equals(GraqlToken.Char.QUOTE) &&
+                literalLastSymbol != null && literalLastSymbol.equals(GraqlToken.Char.QUOTE)) {
+                // Remove surrounding quotes
+                return unquoteString(string);
+            }
+        }
+        throw GraqlException.of(PARSING_ERROR_QUOTEDSTRING_TOKEN.message(string.getText()));
```

Comment:
We had a similar check for Boolean, so I added it for all the other getters for consistency

Reply:
Convert to `assertion`, something like:
```
String str = string.getText();
String start = str.subString(0, 1);
String end = str.substring(string.getText().length() - 1);
assert (Objects.equals(start, GraqlToken.Char.QUOTE.toString() && ..) || ...
```

Reply:
done